documentation/src/main/asciidoc/topics/ref_persistence_spi.adoc

```diff
@@ -0,0 +1,48 @@
+
+link:{javadocroot}/org/infinispan/persistence/spi/CacheWriter.html[CacheWriter] and link:{javadocroot}/org/infinispan/persistence/spi/CacheLoader.html[CacheLoader]::
+Provide basic methods for writing to and reading from cache stores.
+
+link:{javadocroot}/org/infinispan/persistence/spi/AdvancedCacheLoader.html[AdvancedCacheLoader] and link:{javadocroot}/org/infinispan/persistence/spi/AdvancedCacheWriter.html[AdvancedCacheWriter]::
+Provide bulk operations to manipulate the underlaying storage, such as parallel
+iteration and purging of expired entries, clear and size.
+
+link:{javadocroot}/org/infinispan/persistence/spi/SegmentedAdvancedLoadWriteStore.html[SegmentedAdvancedLoadWriteStore]::
+Provides all the operations that deal with segments.
+
+Cache stores can be segmented if they do one of the following:
+
+* Implement the
+link:{javadocroot}/org/infinispan/persistence/spi/SegmentedAdvancedLoadWriteStore.html[SegmentedAdvancedLoadWriteStore] interface. In this case only a single
+store instance is used per cache.
+
+* Has a configuration that extends the link:{javadocroot}/org/infinispan/configuration/cache/AbstractSegmentedConfiguration.html[AbstractSegmentedConfiguration] abstract class.
++
+This requires you to implement the `newConfigurationFrom()` method where it is expected that a new `StoreConfiguration` instance is created per invocation. This creates a store instance per segment to which a node can write. Stores might start and stop as data is moved between nodes.
+
+A provider might choose to only implement a subset of these interfaces:
+
+* Not implementing the  link:{javadocroot}/org/infinispan/persistence/spi/AdvancedCacheWriter.html[AdvancedCacheWriter] makes the given writer not usable for purging expired entries or clear
+
+* If a loader does not implement the link:{javadocroot}/org/infinispan/persistence/spi/AdvancedCacheLoader.html[AdvancedCacheLoader] interface, then it will not participate in preloading nor in cache iteration (required also for stream operations).
+
+//Community only
+ifndef::productized[]
+
+If you want to migrate your existing store to the new API or to write a new store implementation, the link:https://github.com/infinispan/infinispan/blob/master/core/src/main/java/org/infinispan/persistence/file/SingleFileStore.java[SingleFileStore] might be a good starting point/example.
+
+.Additional implementations
+
+You can find cache loader and cache store implementations at link:http://infinispan.org/cache-store-implementations[cache-store-implementations].
```

Comment:
We may want to reconsider the "Extra Cache Stores" section on this page. Most of them have links that don't work and are very out of date at this point. Plus I really don't think we should be pointing at a Redis store \cc @tristantarrant